returnn/util/file_cache.py

```diff
@@ -312,6 +326,26 @@ def _check_existing_copied_file_maybe_cleanup(src_filename: str, dst_filename: s
         return True
 
 
+def _copy_with_prealloc(src: str, dst: str):
+    dst_size = os.stat(src).st_size
+    with open(dst, "wb") as dst_file:
+        if dst_size > 0:
+            # We prealloc size + 1 to avoid having a file that has the exact same
+            # size as its original. After copying, we then truncate the additional
+            # byte away. This allows detecting incompletely copied files.
+            #
+            # See also `_check_existing_copied_file_maybe_cleanup`.
+            if os.name == "posix":
+                os.posix_fallocate(dst_file.fileno(), 0, dst_size + 1)
```

Comment:
Actually, I realize now, we anyway copy to `dst_tmp_filename`, not directly to the target file, so maybe the `dst_size + 1` was not necessary?

Reply:
yeah perhaps, but still doesn't hurt

Reply:
Ok but you should extend the comment then, otherwise someone might get confused and wonder about this.

Reply:
there is a comment on lines 337-341

Reply:
Yea but it is relevant, as I would otherwise think this is a bug in the code and unnecessary. I actually got confused that we do this now as it's not really needed.

Reply:
I extended the docstring in the meantime.

Reply:
Yea but it does not say anything about that this is actually the temp file, so that the `dst_size + 1` is not necessary, and why we still do it anyway.